### PR TITLE
Upgrade SDK to Tekton Pipelines v0.15.0

### PR DESCRIPTION
**Pull request by ckadner**
**Description of your changes:**

* Update SDK, `README`s, `Makefile`, and E2E tests for [Tekton Pipelines v0.15.0](https://github.com/tektoncd/pipeline/releases/tag/v0.15.0)

**Environment tested:**

* Python Version (use `python --version`): `3.7.5`
* Tekton Version (use `tkn version`): `v0.15.0`
* Kubernetes Version (use `kubectl version`): `v1.16.13+IKS`
* OS (e.g. from `/etc/os-release`): Mac OS Catalina

/cc @Tomcli

### CODE REVIEW

**Comment by Tomcli** [comment]
Sorry I just figured out the context variables I want to use is still in master since it didn't make it to the 0.15 branch. 0.15 has several bug fixes that are nice to have (such as variable substitution for finally). How can we convey the message that it can still be used on 0.14 but we will start testing with 0.15.0?

File: sdk/README.md [hunk 1]
```diff
@@ -50,7 +50,7 @@ SDK provides a `TektonCompiler` and a `TektonClient`:
 ## Project Prerequisites
 
  - Python: `3.5.3` or later
- - Tekton: [`0.14.0`](https://github.com/tektoncd/pipeline/releases/tag/v0.14.0)
+ - Tekton: [`0.14.0`](https://github.com/tektoncd/pipeline/releases/tag/v0.15.0) or later
```
Comment (Tomcli) [inline sdk/README.md:53]: Maybe we should have the release page instead of pointing to a certain tag?

Comment (ckadner) [reply sdk/README.md:53]: I want people who already have a Tekton installation to find that their version 0.14 is okay to use, or, if they don't have an installation yet, to follow the link and get the latest version we tested with (0.15.0 now)

Comment (animeshsingh) [reply sdk/README.md:53]: this is confusing though - 0.14 hyperlinked to 0.15 tag. 
Can we say something like supported versions: 0.14 and 0.15, with each hyperlinked to their respective release tags...?

Comment (ckadner) [reply sdk/README.md]: How about:

**Project Prerequisites**
- ...
- Tekton: [`v0.14.0`](https://github.com/tektoncd/pipeline/releases/tag/v0.14.0) or [later](https://github.com/tektoncd/pipeline/releases/latest)
- ...

Comment (ckadner) [reply sdk/README.md]: I tested all of these version, btw:

```YAML
v0.14.0: OK
v0.14.1: OK
v0.14.2: OK
v0.14.3: OK
v0.15.0: OK
v0.15.1: OK
```

**Pull request merged**